app/models/export/to_io.rb

```diff
@@ -9,13 +9,14 @@ def initialize(relation, output)
 
     def run
       output.puts(CSV.generate_line(self.class::HEADER))
+      cache = {}
```

Comment:
I really like this replacement of a process-long cache with one scoped to the processing of a task -- feels like the right balance between allowing objects to be reused over a short time period without running into consistency errors on editing.